drivers/at86rf2xx/at86rf2xx.c

```diff
@@ -136,22 +136,13 @@ size_t at86rf2xx_send(at86rf2xx_t *dev, uint8_t *data, size_t len)
 
 void at86rf2xx_tx_prepare(at86rf2xx_t *dev)
 {
-    uint8_t state;
-
     dev->pending_tx++;
 
-    /* make sure ongoing transmissions are finished */
-    do {
-        state = at86rf2xx_get_status(dev);
-    } while (state == AT86RF2XX_STATE_BUSY_RX_AACK ||
-             state == AT86RF2XX_STATE_BUSY_TX_ARET);
-
+    uint8_t state = at86rf2xx_set_state(dev, AT86RF2XX_STATE_TX_ARET_ON);
     if (state != AT86RF2XX_STATE_TX_ARET_ON) {
```

Comment:
this check was originally deleted in #7115 and is the root cause of the regression, that #7275 fixed - but but plain reversion without exploiting the new behaviour of `at86rf2xx_set_state`.